### DISCUSSION TITLE
How do you put TabBar into a stack navigation's header?

### DISCUSSION THREAD

**Discussion by likeSo** [react-navigation/react-navigation#10252] [General]
How do you achieve this kink of UI?
![image](https://user-images.githubusercontent.com/12247340/148872838-0734a26d-d3c3-4077-94e3-00e470180f83.png)
1. Normally, if a tab navigator is nested into a stack navigator, the tab bar is under the navigation bar;
2. I can't use `navigation.setOptions` because the tab navigator's tab bar is can't be extracted from the tab navigator;
3. If a hide the stack header, then I have to customize the tab bar itself because i need the back icon...

My question:
- **How do you achieve this?**
- Does react-navigation have a solution for this?

**Comment by raajnadar** [comment]
There is not predefined navigation for this but you can create a custom navigator.

https://reactnavigation.org/docs/custom-navigators/

**Reply by likeSo** [reply]
Thank you, but could you tell me what's the difference between customizing header and custom navigator?
Is the customing navigator on structure-layer or UI-layer?

**Reply by ted-shinner** [reply]
i did a workaround for this,
<Stack.Navigator>
  <Stack.screen 
    name="pagename"
    options={{headerTitle: () => null,}}
  />

then in the tab.navigator i used position absolute and negative margins with a zIndex to get that result
